vip-api/src/main/java/fr/insalyon/creatis/vip/api/SpringSecurityConfig.java

```diff
@@ -57,57 +80,95 @@
  * The custom api key configuration is done in {@link ApikeyAuthentificationConfigurer}
  *
  * Created by abonnet on 7/22/16.
+ *
+ * Modified by khalilkes to implement keycloak adapter
  */
+@ComponentScan(basePackageClasses = {KeycloakSecurityComponents.class})
+@KeycloakConfiguration
 @EnableWebSecurity
-public class SpringSecurityConfig extends WebSecurityConfigurerAdapter {
+public class SpringSecurityConfig extends KeycloakWebSecurityConfigurerAdapter {
 
     // authentication done by bean LimitigDaoAuthenticationProvider
 
-    private final ApikeyAuthenticationEntryPoint apikeyAuthenticationEntryPoint;
 
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final ApikeyAuthenticationEntryPoint apikeyAuthenticationEntryPoint;
     private final Environment env;
+    private ApiBusiness apiBusiness;
 
     @Autowired
-    public SpringSecurityConfig(ApikeyAuthenticationEntryPoint apikeyAuthenticationEntryPoint, Environment env) {
+    public SpringSecurityConfig(ApikeyAuthenticationEntryPoint apikeyAuthenticationEntryPoint, Environment env, ApiBusiness apiBusiness) {
         this.apikeyAuthenticationEntryPoint = apikeyAuthenticationEntryPoint;
         this.env = env;
+        this.apiBusiness = apiBusiness;
+    }
+
+
+    @Override
+    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
+        return new RegisterSessionAuthenticationStrategy(new SessionRegistryImpl());
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) {
+        KeycloakAuthenticationProvider keycloakAuthenticationProvider = keycloakAuthenticationProvider();
+        keycloakAuthenticationProvider.setGrantedAuthoritiesMapper(grantedAuthoritiesMapper());
+        auth.authenticationProvider(keycloakAuthenticationProvider);
     }
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
+        super.configure(http);
         http
-            .authorizeRequests()
+                .authorizeRequests()
                 .antMatchers("/rest/platform").permitAll()
                 .antMatchers("/rest/authenticate").permitAll()
+                .antMatchers("/rest/register").authenticated() //signup a user to VIP
```

Comment:
Good it is secured, but who will call this method ? I thought it will be called by shanoir-server itself and should be secured by IP or by a specific keycloak link.

Reply:
I didn't know where to add the property, so i added it in `vip-api.conf` , while adding it in `ApiPropertyIntialize` class to verify the property ...